Sources/SourceKitBazelBSP/Server/BaseServerConfig.swift

```diff
@@ -29,11 +29,36 @@ package struct BaseServerConfig: Equatable {
     let targets: [String]
     let indexFlags: [String]
     let filesToWatch: String?
+    let autoDiscoverTargets: Bool
 
-    package init(bazelWrapper: String, targets: [String], indexFlags: [String], filesToWatch: String?) {
+    package init(
+        bazelWrapper: String,
+        targets: [String],
+        indexFlags: [String],
+        filesToWatch: String?,
+        autoDiscoverTargets: Bool = true
+    ) {
         self.bazelWrapper = bazelWrapper
         self.targets = targets
         self.indexFlags = indexFlags
         self.filesToWatch = filesToWatch
+        self.autoDiscoverTargets = autoDiscoverTargets
+    }
+
+    /// Returns the resolved targets, auto-discovering them if needed and enabled
+    package func resolvedTargets(
```

Comment:
Suggestion: Instead of having the helper in this file and resolving this in `BazelTargetQuerier.swift`, I was thinking we could try run this directly in `Serve.swift`. That way you can pass the resolved targets directly to `targets`, thus not requiring the additional logic here / on the other requests. WDYT?